source/tls_channel_handler.c

```diff
@@ -272,8 +264,27 @@ int aws_tls_ctx_options_init_client_mtls_with_pkcs11(
         aws_string_destroy(pkcs_private_key_object_label);
     }
 
+    struct aws_byte_buf tmp_cert_buf;
+    if (aws_byte_buf_init(&tmp_cert_buf, allocator, 0) != AWS_OP_SUCCESS) {
+        AWS_LOGF_ERROR(AWS_LS_IO_TLS, "Could not allocate byte buffer for custom key operation certificate");
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        goto error;
+    }
+    if (pkcs11_options->cert_file_contents.ptr == NULL) {
+        AWS_LOGF_ERROR(AWS_LS_IO_TLS, "Certificate file contents are empty");
+        return AWS_OP_ERR;
+    } else {
+        if (aws_byte_buf_init_copy_from_cursor(&tmp_cert_buf, allocator, pkcs11_options->cert_file_contents)) {
+            AWS_LOGF_ERROR(AWS_LS_IO_TLS, "Could not allocate byte buffer for custom key operation certificate");
+            return AWS_OP_ERR;
```

Comment:
this function allocates temporary things.
you can't just return AWS_OP_ERR if things go wrong, you need to goto